Updated documentation and added conversion utility (#34319)
huggingface/transformers: docs/source/en/tiktoken.md

model_id = "meta-llama/Meta-Llama-3-8B-Instruct"
tokenizer = AutoTokenizer.from_pretrained(model_id, subfolder="original") 
```

> 
> The `tokenizer.model` file can be generated with `dump_tiktoken_bpe` function from `tiktoken.load`.
> 
> However, keep in mind that `tokenizer.model` file contains no information about additional tokens or a pattern string. 
> If these are important for you, consider converting the tokenizer to HF format and saving it on disk using the utility 
> function:

Reviewer: Maybe we can create a new section here that describes the purpose and motivation a little more.

```suggestion
## Create tiktoken tokenizer

The `tokenizer.model` file contains no information about additional tokens or pattern strings. If these are important, convert the tokenizer to `tokenizer.json`, the appropriate format for [`PreTrainedTokenizerFast`].

Generate the `tokenizer.model` file with [tiktoken.get_encoding](https://github.com/openai/tiktoken/blob/63527649963def8c759b0f91f2eb69a40934e468/tiktoken/registry.py#L63) and then convert it to `tokenizer.json` with [`convert_tiktoken_to_fast`].
```

```py
from transformers.tokenization_utils_fast import convert_tiktoken_to_fast
from tiktoken import get_encoding

# You can load your custom encoding or the one provided by OpenAI
encoding = get_encoding("gpt2")
convert_tiktoken_to_fast(encoding, "config/save/dir")
```

> This will convert tiktoken tokenizer to `PretrainedTokenizerFast` and save its configuration file, `tokenizer.json`, to
> provided directory.
> 
> You can load this tokenizer with `PretrainedTokenizerFast.from_pretrained("config/save/dir")`.

Reviewer: ```suggestion
The resulting `tokenizer.json` file is saved to the specified directory and loaded with [`PreTrainedTokenizerFast`].

```py
PreTrainedTokenizerFast.from_pretrained("config/save/dir")
```
```
